src/os/bluestore/BlueStore.h

```diff
@@ -295,17 +298,21 @@ class BlueStore : public ObjectStore,
     uint16_t cache_private = 0; ///< opaque (to us) value used by Cache impl
     uint32_t flags;             ///< FLAG_*
     uint64_t seq;
-    uint32_t offset, length;
+    uint64_t offset, length;
```

Comment:
There is no need to go 64 bits.
Max object size in BlueStore is 32 bits.
I interpret that in the way, that the offset of the last accessible byte of object is < 2^32.

https://github.com/ceph/ceph/blob/main/src/os/bluestore/BlueStore.cc#L145